### PR TITLE
feat(master-detail): added master-detail layout updates

### PR DESCRIPTION
**Pull request by mattnolting**
closes: #2256

### CODE REVIEW

File: src/patternfly/components/Drawer/examples/Drawer.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ cssPrefix: pf-c-drawer
 ---
 
 ## Examples
+
 ```hbs title=Closed
```
Comment (christiemolloy) [inline src/patternfly/components/Drawer/examples/Drawer.md]: Should this be "Closed panel on right", then "Expanded panel on right", then "Closed panel on left" and then "Expanded panel on left"

File: src/patternfly/components/Drawer/drawer.scss [hunk 2]
```diff
@@ -21,55 +33,194 @@
...
+      transform: translateX(-100%);
+    }
+  }
+
+  // Panel left
+  .pf-c-drawer.pf-m-panel-left {
+    .pf-c-drawer__panel {
+      order: 0;
+      margin-right: calc(var(--pf-c-drawer__panel--FlexBasis) * -1);
+      transform: translateX(-100%);
```
Comment (christiemolloy) [inline src/patternfly/components/Drawer/drawer.scss]: Should these `translateX` properties use variables? You did for `transition` so might help to be consistent unless there's a reason why

File: src/patternfly/components/Drawer/drawer.scss [hunk 3]
```diff
@@ -21,55 +33,194 @@
...
+    transform: translateX(0);
+  }
+}
+
+// Drawer inline
+@media screen and (min-width: $pf-global--breakpoint--md) {
+  .pf-c-drawer.pf-m-inline {
+    // Inline
+    .pf-c-drawer__panel {
+      margin-left: calc(var(--pf-c-drawer__panel--FlexBasis) * -1);
```
Comment (christiemolloy) [inline src/patternfly/components/Drawer/drawer.scss]: should this value for `margin-left` be a variable?

File: src/patternfly/components/Drawer/drawer.scss [hunk 4]
```diff
@@ -21,55 +33,194 @@
...
+
+    // Inline, expanded
+    &.pf-m-expanded .pf-c-drawer__panel {
+      margin-left: 0;
+      transform: translateX(0);
+    }
+
+    // Inline, panel left
+    &.pf-m-panel-left .pf-c-drawer__panel {
+      margin-right: calc(var(--pf-c-drawer__panel--FlexBasis) * -1);
```
Comment (christiemolloy) [inline src/patternfly/components/Drawer/drawer.scss]: should this value for `margin-right` be a variable?

File: src/patternfly/components/Drawer/drawer.scss (hunk 2 above)
Comment (mattnolting) [reply src/patternfly/components/Drawer/drawer.scss]: No, I don't think so as this value should not change.

File: src/patternfly/components/Drawer/drawer.scss (hunk 3 above)
Comment (mattnolting) [reply src/patternfly/components/Drawer/drawer.scss]: Same as above

File: src/patternfly/components/Drawer/drawer.scss (hunk 4 above)
Comment (mattnolting) [reply src/patternfly/components/Drawer/drawer.scss]: Same ^^

File: src/patternfly/components/Drawer/drawer.scss (hunk 2 above)
Comment (mattnolting) [reply src/patternfly/components/Drawer/drawer.scss]: `margin-right: calc(var(--pf-c-drawer__panel--FlexBasis) * -1);` has to stay as is for the margining to work properly for `.pf-c-drawer.pf-m-inline .pf-c-drawer__panel.pf-m-width-{mod}` to function properly.

File: src/patternfly/components/Drawer/examples/Drawer.md (hunk 1 above)
Comment (mattnolting) [reply src/patternfly/components/Drawer/examples/Drawer.md]: Updated

Comment (mattnolting) [reply src/patternfly/components/Drawer/examples/Drawer.md]: Updated

File: src/patternfly/components/Drawer/drawer.scss [hunk 5]
```diff
@@ -1,16 +1,28 @@
+$pf-c-drawer--breakpoint-map: build-breakpoint-map("base", "lg", "xl", "2xl");
+$pf-c-drawer__panel--width-list: (25, 33, 50, 66, 75);
```
Comment (christiemolloy) [inline src/patternfly/components/Drawer/drawer.scss]: would someone want a drawer that comes out all the way to 100%?

File: src/patternfly/components/Drawer/drawer.scss [hunk 6]
```diff
@@ -21,55 +33,195 @@
...
+  padding: var(--pf-c-drawer__content-body--m-padding--PaddingTop) var(--pf-c-drawer__content-body--m-padding--PaddingRight) var(--pf-c-drawer__content-body--m-padding--PaddingBottom) var(--pf-c-drawer__content-body--m-padding--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body {
+  padding: var(--pf-c-drawer__panel-body--PaddingTop) var(--pf-c-drawer__panel-body--PaddingRight) var(--pf-c-drawer__panel-body--PaddingBottom) var(--pf-c-drawer__panel-body--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body.pf-m-no-padding {
```
Comment (mcoker) [inline src/patternfly/components/Drawer/drawer.scss]: can this nest in the selector above with `&.pf-m-no-padding`?

File: src/patternfly/components/Drawer/drawer.scss [hunk 7]
```diff
@@ -21,55 +33,195 @@
...
+
+// Expanded
+.pf-c-drawer.pf-m-expanded .pf-c-drawer__panel {
+  transform: translateX(-100%);
+}
+
+// Layout modifications happen after md breakpoint
+@media screen and (min-width: $pf-global--breakpoint--md) {
+  // Default
+  .pf-c-drawer:not(.pf-m-divider) .pf-c-drawer__panel {
```
Comment (mcoker) [inline src/patternfly/components/Drawer/drawer.scss]: You could move this down into the `.pf-c-drawer {}` block below with `&:not(.pf-m-divider) .pf-c-drawer__panel`.

File: src/patternfly/components/Drawer/drawer.scss [hunk 8]
```diff
@@ -21,55 +33,195 @@
...
 }
 
-.pf-c-drawer__content {
-  z-index: var(--pf-c-drawer__content--ZIndex);
-  flex-grow: 1;
-  min-width: 0;
-  margin-right: var(--pf-c-drawer__content--MarginRight);
-  transition: var(--pf-c-drawer__content--Transition);
+// Drawer inline
+@media screen and (min-width: $pf-global--breakpoint--md) {
```
Comment (mcoker) [inline src/patternfly/components/Drawer/drawer.scss]: could you remove this media query and just move the contents into the previous one?

File: src/patternfly/components/Drawer/drawer.scss [hunk 9]
```diff
@@ -21,55 +33,195 @@
...
-  z-index: var(--pf-c-drawer__panel--ZIndex);
-  width: var(--pf-c-drawer__panel--Width);
-  overflow-x: hidden;
-  overflow-y: auto;
-  background-color: var(--pf-c-drawer__panel--BackgroundColor);
-  transition: var(--pf-c-drawer__panel--Transition);
-  transform: var(--pf-c-drawer__panel--Transform);
-  -webkit-overflow-scrolling: touch;
+// Panel divider
+@media screen and (min-width: $pf-global--breakpoint--md) {
```
Comment (mcoker) [inline src/patternfly/components/Drawer/drawer.scss]: same here - could we just have all of the `--md` styles in a single breakpoint?

**Pull request merged**